lib/fog/azurerm/requests/network/attach_network_security_group_with_subnet.rb

```diff
@@ -0,0 +1,59 @@
+module Fog
+  module Network
+    class AzureRM
+      # Real class for Network Request
+      class Real
+        def attach_network_security_group_with_subnet(resource_group, subnet_name, virtual_network_name, address_prefix, route_table_id, network_security_group_id)
```

Comment:
- 'with' renamed to 'to'
- Argument order discussed

- whats the use of route_table_id here?
We need to pass old value of route table id here because we want to keep the rout table. If we don't pass route_table_id then it will be detached. The reason is that we send the update call to api.